c2rust-analyze/src/rewrite/expr/convert.rs

```diff
@@ -286,6 +286,126 @@ impl<'tcx> ConvertVisitor<'tcx> {
                 )
             }
 
+            mir_op::RewriteKind::MallocSafe {
+                ref zero_ty,
+                elem_size,
+                single,
+            } |
+            mir_op::RewriteKind::CallocSafe {
+                ref zero_ty,
+                elem_size,
+                single,
+            } => {
+                // `malloc(n)` -> `Box::new(z)` or similar
```

Comment:
Could you add an assertion here that the size of `zero_ty` is `elem_size`? Or can `zero_ty` be a different type?

Reply:
The sizes don't necessarily match.  `elem_size` is the size of the original, unrewritten type, which we use for converting `memcpy`/`memset` byte counts to a number of elements.  `zero_ty` indicates how to zero out the rewritten type, which might have a different size.  (Also, `ZeroizeTy` doesn't actually contain enough detail to allow computing the size.)

Reply:
> `zero_ty` indicates how to zero out the rewritten type, which might have a different size.

Could that cause compiler errors?

Reply:
No, but there's a related issue of potentially using the wrong size when converting the byte count passed to `malloc`/`memcpy`/`memset` to an element count.  I have a fix for this (at least for the most common cases) in a later branch: fd24d8158dbd7e4f0900661ae8a6afe4bc4bbc38